beets/autotag/hooks.py

```diff
@@ -16,236 +16,201 @@
-    is equivalent to `d['field']`.
-    """
+    """Mapping enabling attribute-style access to stored metadata values."""
+
+    def copy(self) -> Self:
+        return deepcopy(self)
 
     def __getattr__(self, attr: str) -> V:
         if attr in self:
             return self[attr]
-        else:
-            raise AttributeError
 
-    def __setattr__(self, key: str, value: V):
+        raise AttributeError(
+            f"'{self.__class__.__name__}' object has no attribute '{attr}'"
+        )
+
+    def __setattr__(self, key: str, value: V) -> None:
         self.__setitem__(key, value)
 
-    def __hash__(self):
+    def __hash__(self) -> int:  # type: ignore[override]
         return id(self)
 
 
-class AlbumInfo(AttrDict[Any]):
-    """Describes a canonical release that may be used to match a release
-    in the library. Consists of these data members:
+class Info(AttrDict[Any]):
+    """Container for metadata about a musical entity."""
+
+    def __init__(
+        self,
+        album: str | None = None,
+        artist_credit: str | None = None,
+        artist_id: str | None = None,
+        artist: str | None = None,
+        artists_credit: list[str] | None = None,
+        artists_ids: list[str] | None = None,
+        artists: list[str] | None = None,
+        artist_sort: str | None = None,
+        artists_sort: list[str] | None = None,
+        data_source: str | None = None,
+        data_url: str | None = None,
+        genre: str | None = None,
+        media: str | None = None,
+        **kwargs,
+    ) -> None:
+        self.album = album
+        self.artist = artist
+        self.artist_credit = artist_credit
+        self.artist_id = artist_id
+        self.artists = artists or []
+        self.artists_credit = artists_credit or []
+        self.artists_ids = artists_ids or []
+        self.artist_sort = artist_sort
+        self.artists_sort = artists_sort or []
+        self.data_source = data_source
+        self.data_url = data_url
+        self.genre = genre
+        self.media = media
+        self.update(kwargs)
+
 
-    - ``album``: the release title
-    - ``album_id``: MusicBrainz ID; UUID fragment only
-    - ``artist``: name of the release's primary artist
-    - ``artist_id``
-    - ``tracks``: list of TrackInfo objects making up the release
+class AlbumInfo(Info):
+    """Metadata snapshot representing a single album candidate.
 
-    ``mediums`` along with the fields up through ``tracks`` are required.
-    The others are optional and may be None.
+    Aggregates track entries and album-wide context gathered from an external
+    provider. Used during matching to evaluate similarity against a group of
+    user items, and later to drive tagging decisions once selected.
     """
 
-    # TYPING: are all of these correct? I've assumed optional strings
     def __init__(
         self,
         tracks: list[TrackInfo],
-        album: str | None = None,
+        *,
```

Comment:
What is this star even doing here? Unpacking and disregarding all args? Could you explain why it was added?